src/main/java/de/hpi/bpt/argos/api/eventType/EventTypeEndpointImpl.java

```diff
@@ -1,93 +1,371 @@
+                (Request request, Response response) ->
+                        endpointUtil.executeRequest(logger, request, response, this::getEventTypes));
+
+        sparkService.get(EventTypeEndpoint.getEventTypeBaseUri(),
+				(Request request, Response response) ->
+						endpointUtil.executeRequest(logger, request, response, this::getEventType));
+
+        sparkService.post(EventTypeEndpoint.getCreateEventTypeBaseUri(),
+				(Request request, Response response) ->
+						endpointUtil.executeRequest(logger, request, response, this::createEventType));
+
+        sparkService.delete(EventTypeEndpoint.getDeleteEventTypeBaseUri(),
+				(Request request, Response response) ->
+						endpointUtil.executeRequest(logger, request, response, this::deleteEventType));
+
+        sparkService.get(EventTypeEndpoint.getEventTypeAttributesBaseUri(),
+				(Request request, Response response) ->
+						endpointUtil.executeRequest(logger, request, response, this::getEventTypeAttributes));
+
+        sparkService.get(EventTypeEndpoint.getEventTypeQueriesBaseUri(),
+				(Request request, Response response) ->
+						endpointUtil.executeRequest(logger, request, response, this::getEventTypeQueries));
+
+        sparkService.get(EventTypeEndpoint.getEventTypeEntityMappingsBaseUri(),
+				(Request request, Response response) ->
+						endpointUtil.executeRequest(logger, request, response, this::getEventTypeEntityMappings));
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getEventTypes(Request request, Response response) {
+        List<EventType> eventTypes = PersistenceAdapterImpl.getInstance().getEventTypes();
+        JsonArray jsonEventTypesArray = new JsonArray();
+
+        for (EventType eventType : eventTypes) {
+            jsonEventTypesArray.add(RestEndpointCommon.getEventTypeJson(eventType));
+        }
+
+        return serializer.toJson(jsonEventTypesArray);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getEventType(Request request, Response response) {
+        long eventTypeId = getEventTypeId(request);
+        EventType eventType = PersistenceAdapterImpl.getInstance().getEventType(eventTypeId);
+        JsonObject jsonEventType = RestEndpointCommon.getEventTypeJson(eventType);
+
+        return serializer.toJson(jsonEventType);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String createEventType(Request request, Response response) {
+		JsonObject jsonEventType = jsonParser.parse(request.body()).getAsJsonObject();
+
+		if (jsonEventType == null) {
+			halt(HttpStatusCodes.BAD_REQUEST, "no event type given in body");
+		}
+
+		EventType eventType = createEventTypeFromJson(jsonEventType);
+
+		if (eventType == null) {
+			halt(HttpStatusCodes.BAD_REQUEST, "event type name already in use, or failed to parse event type");
+		}
+
+		PersistenceAdapterImpl.getInstance().createArtifact(eventType, EventTypeEndpoint.getEventTypeUri(eventType.getId()));

```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png 'Severity: BLOCKER') NullPointerException might be thrown as 'eventType' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS2259)